pkg/parse/run.go

```diff
@@ -78,6 +78,12 @@ type RunFunc func(ctx context.Context, p Parser, trigger string, state *reconcil
 
 // Run keeps checking whether a parse-apply-watch loop is necessary and starts a loop if needed.
 func Run(ctx context.Context, p Parser, nsControllerState *namespacecontroller.State, runOpts RunOpts) {
+	// Pre-initialize count metrics to ensure visibility, even if errors haven't occurred.
+	// Special label indicates initial state. Metrics are cumulative and used for rate aggregation
+	// to detect ongoing errors. Zero value means no recent errors.
+	metrics.RecordResourceFight(ctx, metrics.Initialization)
+	metrics.RecordInternalError(ctx, metrics.Initialization)
+	metrics.RecordResourceConflict(ctx, metrics.Initialization)
```

Comment:
I think the initial intention was to make these metrics appear no matter the error conditions have happened or not, otherwise the metrics will be missing until the error condition has happened at least once. And yes, recording them will plus 1 to the counter value.

Reply:
This will make these metrics all blip to a rate > 0 whenever the reconciler starts. I don't know if that's a great tradeoff...

Is there a way to set the value to zero and then use a rate query that only matches value=1?

Reply:
Yes correct, with this method the rate will go up then resume 0 when reconciler starts. I think we've been through this: once we `set`, it will increment the metric value by 1 whether we set a 0 or not. We could fallback to previous when we tolerate the metric being missing. Or it's a bigger change, we make it a lastValue and give it a meaningful count.

Reply:
This part has been moved to https://github.com/GoogleContainerTools/kpt-config-sync/pull/1237, will be making change accordingly.